src/modules/sway/bar.cpp

```diff
@@ -61,38 +66,108 @@ void BarIpcClient::onInitialConfig(const struct Ipc::ipc_response& res) {
 void BarIpcClient::onIpcEvent(const struct Ipc::ipc_response& res) {
   try {
     auto payload = parser_.parse(res.payload);
-    if (auto id = payload["id"]; id.isString() && id.asString() != bar_.bar_id) {
-      spdlog::trace("swaybar ipc: ignore event for {}", id.asString());
-      return;
-    }
-    if (payload.isMember("visible_by_modifier")) {
-      // visibility change for hidden bar
-      signal_visible_(payload["visible_by_modifier"].asBool());
-    } else {
-      // configuration update
-      auto config = parseConfig(payload);
-      signal_config_(std::move(config));
+    switch (res.type) {
+      case static_cast<uint32_t>(IPC_EVENT_WORKSPACE):
+        if (payload.isMember("change")) {
+          // only check and send signal if the workspace update reason was because of a urgent
+          // change
+          if (payload["change"] == "urgent") {
+            // Retrieve status for all workspace in order to check if urgency is still active for at
+            // least one workspace
+            ipc_.sendCmd(IPC_GET_WORKSPACES);
+          }
+          last_press_modifier_ = false;
```

Comment:
You are changing the `last_press_modifier_` directly from the ipc worker thread. It would be preferable to use `std::atomic<bool>` or signal.

Reply:
Sorry, might have misunderstood your explanation before, I thought it will only be possible to write to this at one point.
Added the atomic, not sure if just changing the type was enough, though?

Reply:
Yep, changing the type is enough. [`std::atomic<T>::operator=`](https://en.cppreference.com/w/cpp/atomic/atomic/operator%3D) is overloaded to take care of atomic assignments when we don't need to swap or check the current value.

> I thought it will only be possible to write to this at one point.

Change to non-atomic variable in one thread may be not (immediately) visible to a read operation in another thread; this behavior is undefined in the C++ memory model. That's why I insisted on some kind of synchronization.